### PR TITLE
[Validation] Display notice if package validation exceeds expectation

### PR DESCRIPTION
**Pull request by loic-sharma**
Displays a notice on the Display Package page if the validation is taking longer than expected.

validationtoolong

Progress on https://github.com/NuGet/Engineering/issues/1144

### CODE REVIEW

File: src/NuGetGallery/Configuration/IAppConfiguration.cs [hunk 1]
```diff
@@ -254,6 +259,11 @@ public interface IAppConfiguration : ICoreMessageServiceConfiguration
         /// </summary>
         bool IsODataFilterEnabled { get; set; }
 
+        /// <summary>
+        /// Gets/sets a string that is a link to the status page
+        /// </summary>
+        string ExternalStatusUrl { get; set; }
```
Comment (joelverhagen) [inline src/NuGetGallery/Configuration/IAppConfiguration.cs]: 👍

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml [hunk 2]
```diff
@@ -153,6 +153,15 @@
                         validation and indexing may take up to an hour. <a href="https://aka.ms/NuGetPackageValidation">Read more</a>.
                     </text>
                 )
+
+                if (Model.ValidatingTooLong)
+                {
+                    @ViewHelpers.AlertWarning(
+                        @<text>
+                            <strong>Package validation is taking longer than expected.</strong> This is not normal, but we are on it!
```
Comment (joelverhagen) [inline src/NuGetGallery/Views/Packages/DisplayPackage.cshtml]: Should this mention the status URL?

File: src/NuGetGallery/Configuration/IAppConfiguration.cs (hunk 1 above)
Comment (joelverhagen) [inline src/NuGetGallery/Configuration/IAppConfiguration.cs]: Also, can we configure this to the right values on DEV and INT? (nitpick)

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml (hunk 2 above)
Comment (loic-sharma) [reply src/NuGetGallery/Views/Packages/DisplayPackage.cshtml]: @anangaur asked that we don't show the status URL until we have automatic status page updates, so I removed the link.

File: src/NuGetGallery/Configuration/IAppConfiguration.cs (hunk 1 above)
Comment (loic-sharma) [reply src/NuGetGallery/Configuration/IAppConfiguration.cs]: Sure! That'll be done in the private repo.

Comment (loic-sharma) [reply src/NuGetGallery/Configuration/IAppConfiguration.cs]: Fixed in [this change](https://nuget.visualstudio.com/NuGetMicrosoft/_git/NuGetDeployment/pullrequest/349?_a=files&iteration=3&base=2).

File: src/NuGetGallery/Views/Shared/Gallery/Footer.cshtml [hunk 3]
```diff
@@ -8,7 +8,7 @@
                 </p>
             </div>
             <div class="col-sm-4">
-                <span class="footer-heading"><a href="https://status.nuget.org/">Status</a></span>
+                <span class="footer-heading"><a href="@this.Config.Current.ExternalStatusUrl">Status</a></span>
```
Comment (chenriksson) [inline src/NuGetGallery/Views/Shared/Gallery/Footer.cshtml]: yay!

**Pull request merged**